src/bundles/files.js

```diff
@@ -361,9 +349,23 @@ export default (opts = {}) => {
         content: pageContent.content.sort((a, b) => {
           if (a.type === b.type || isMac) {
             if (sorting.by === sorts.BY_NAME) {
-              return compare(a.name, b.name, sorting.asc)
+              a = a.name
+              b = b.name
+
+              return sorting.asc
+                ? a.localeCompare(b, undefined, { numeric: true, sensitivity: 'base' })
```

Comment:
Sorting can be simpler if we give oursleves the helper functions to hide the details... The "one weird trick" to for comparator functions is to pass the sort direction as 1 or -1 so you can just mulitply the result of the comparing a and b by dir. If (a - b) is a postive but dir is -1, you invert the result, which is what you want when changing the sort order.

```js
// some where else in a sort.js lib file

//  returns a natural sort comparator for strings. `dir` is 1 for ascending or -1 for descending
export function sortByName (dir = 1, opts  = { numeric: true, sensitivity: 'base' }) {
   return (a, b) => a.localeCompare(b, undefined, opts) * dir
}

//  returns a numerical sort comparator. `dir` is 1 for ascending or -1 for descending
export function sortBySize (dir  = 1) {
  return (a, b) => (a - b) * dir
}
  
// here at use site
if (sorting.by === sorts.BY_NAME) {
  return sortByName(sorting.asc ? 1 : -1)(a.name, b.name)
} else {
  return sortBySize(sorting.asc ? 1 : -1)(a.cumulativeSize || a.size, b.cumulativeSize || b.size)
}
```

Lets extract this to a helper and reuse it in both locations where we want name based sorting, so future us can see the right way to do it rather than defaulting to ascii-sorting.

Reply:
cc @hacdias do you  know the "one weird trick" when writing comparator functions?

Reply:
Thanks, this is so much better! ✨

Reply:
@olizilla wow. Never thought about that!